tests/workloads-spin-registry-push/workload-spin-registry-push.yaml

```diff
@@ -0,0 +1,212 @@
+apiVersion: apps/v1
```

Comment:
It seems like the OCI image name describes the packaging mechanism (spin registry push vs docker). Should we have one spin-deployments folder and put them all in there?

Reply:
having one yaml for both workloads complicates the setup part (pushing/importing same images built in different format) and can potentially create confusion when debugging failures (which artifact was actually pulled when deploying). therefore I thought to keep them separate.

Reply:
To clarify, i still mean two different yaml files i just don't see the need for two directories

Reply:
got it. 

This is done this way because workload apply script do a `kubectl apply -f <dir>`. if we have both files in same dir, we will need to explicitly add file names when doing `kubectl apply -f` and that would mean to keep that list up-to-date when we add/remove yaml files in that dir.